lib/crowbar/init/application.rb

```diff
@@ -315,6 +315,9 @@ class Application < Sinatra::Base
       # api_version "2.0"
       post "/api/upgrade/new" do
         api_constraint(2.0)
+        upgrade_status = ::Crowbar::UpgradeStatus.new(logger)
+        upgrade_status.start_step
```

Comment:
I would assume something like that. But to be fair, we are not doing such check for any other step. We probably have to update the library method somehow to take care about the order itself.